src/common/strtol.h

```diff
@@ -48,4 +48,20 @@ Target strict_si_cast(const char *str, std::string *err) {
 template <>
 uint64_t strict_si_cast(const char *str, std::string *err);
 
+template<typename T, const unsigned base = 10, const unsigned width = 1>
+static inline
+char* ritoa(T u, char *buf)
+{
+  static_assert(base <= 16, "extend character map below to support higher bases");
+  unsigned digits = 0;
+  while (u) {
+    *--buf = "0123456789abcdef"[u % base];
+    u /= base;
+    digits++;
+  }
+  while (digits++ < width)
+    *--buf = '0';
+  return buf;
+}
+
```

Comment:
This template will also instantiate for signed integer types, and will yield improper conversions. Can it be protected against it?